File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int left=0;
        int right=0;
        HashMap<Integer, Integer> map=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(target-nums[i])){
                right=i;
                left=map.get(target-nums[i]);
            }
            map.put(nums[i],i);
        }
        return new int[]{left,right};
    }
}

File: Data Structures & Algorithms/is-palindrome/submission-1.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int i=0;
        int j=s.Length-1;
        while(i<j){
            while(i<j && !(((s[i]>='a') && (s[i]<='z')) || ((s[i]>='A') && (s[i]<='Z')) || ((s[i]>='0') && (s[i]<='9')))){
                i++;
            }
            while(i<j && !(((s[j]>='a') && (s[j]<='z')) || ((s[j]>='A') && (s[j]<='Z')) || ((s[j]>='0') && (s[j]<='9')))){
                j--;
            }
            char left=s[i];
            char right=s[j];
            if(left>='A' && left<='Z'){
                left=(char)(left+32);
            }
            if(right>='A' && right<='Z'){
                right=(char)(right+32);
            }
            if(left != right){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}
